components/pthread/pthread.c

```diff
@@ -423,11 +455,9 @@ void pthread_exit(void *value_ptr)
     if (!pthread) {
         assert(false && "Failed to find pthread for current task!");
     }
-    if (pthread->task_arg) {
-        free(pthread->task_arg);
-    }
     if (pthread->detached) {
         // auto-free for detached threads
+        vTaskDelete(NULL);
```

Comment:
What was the reason for moving `vTaskDelete(NULL)` here? It seems that the semantics are changed significantly: the following two statements won't be executed, the log call and giving the semaphore below won't be executed, either.